test/functional/tests/cache_ops/test_core_add.py

```diff
@@ -0,0 +1,216 @@
+#
+# Copyright(c) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import os
+from itertools import cycle
+from random import shuffle
+
+import pytest
+
+from api.cas import casadm
+from api.cas.casadm_parser import get_cores
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.fs_utils import parse_ls_output, ls, remove, readlink
+from test_utils.filesystem.symlink import Symlink
+from test_utils.output import CmdException
+from test_utils.size import Unit, Size
+
+cores_number = 4
+by_id_dir = '/dev/disk/by-id/'
+custom_dir = '/tmp/castle'
+symlink_name = 'dinosaur'
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_add_core_path_by_id():
+    """
+    title: Test for adding core with by-id path.
+    description: |
+      Check if OpenCAS accepts by-id path to disks added as cores.
+    pass_criteria:
+      - Cores are added to cache
+      - Cores are added to cache with the same path as given
+    """
+    with TestRun.step("Clearing dmesg"):
+        TestRun.executor.run_expect_success("dmesg -C")
+
+    with TestRun.step("Prepare partitions for cache and for cores."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(200, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(400, Unit.MebiByte)] * cores_number)
+
+    with TestRun.step(
+            f"Create symlinks for {core_dev.path} partitions in {by_id_dir} directory."
+    ):
+        for i, partition in enumerate(core_dev.partitions):
+            Symlink.create_symlink(by_id_dir, partition.path)
+
+    with TestRun.step(f"Find symlinks to {core_dev.path} in {by_id_dir}."):
+        links = [
+            Symlink(os.path.join(by_id_dir, item.full_path))       # parse_ls_output returns
+            for item in parse_ls_output(ls(by_id_dir), by_id_dir)  # symlinks without path
+            if isinstance(item, Symlink)
+        ]
+        core_dev_links = [link for link in links if readlink(core_dev.path) in link.get_target()]
+
+    with TestRun.step(f"Select different links to {core_dev.path} partitions."):
+        selected_links = select_links(core_dev_links)
+
+    with TestRun.step("Start cache and add cores"):
+        paths_from_dmesg = []
+        cache = casadm.start_cache(cache_part, force=True)
+        for i in range(cores_number):
+            core_dev.partitions[i].path = selected_links[i].full_path
+            cache.add_core(core_dev.partitions[i])
+            paths_from_dmesg.append(get_added_core_path_from_dmesg())
```

Comment:
there is no need to parse dmesg. "casadm -L -b" will print paths exactly as returned by OCF, without any translation.